### PR TITLE
Support for comparing objects with boolean type in `isLike` function

### PR DESCRIPTION
**Pull request by monaqa**
Added the ability to compare objects/arrays containing bool types with isLike.

To achieve this, I have added two functions and their corresponding tests.

- `isBoolean(x: unknown): x is boolean` : returns true if and only if `x` is boolean
- `ensureBoolean(x: unknown): asserts x is boolean` : ensures `x` is boolean

### CODE REVIEW

File: is_test.ts [hunk 1]
```diff
@@ -37,6 +38,20 @@ Deno.test("isNumber returns false on non array", () => {
...
+Deno.test("isBoolean returns false on non boolean", () => {
+  assertEquals(isBoolean(0), false);
+  assertEquals(isBoolean(""), false);
+  assertEquals(isBoolean([]), false);
+  assertEquals(isBoolean({}), false);
+  assertEquals(isBoolean(function () {}), false);
+  assertEquals(isBoolean(null), false);
+  assertEquals(isBoolean(undefined), false);
+});
+
```
Comment (lambdalisue) [inline is_test.ts:54]: Please add tests like the below to all other tests.

```typescript
assertEquals(isXXXXX(true), false);
assertEquals(isXXXXX(true), false);
```

File: ensure_test.ts [hunk 2]
```diff
@@ -48,6 +49,20 @@ Deno.test("ensureNumber throws error on non number", () => {
...
+});
+Deno.test("ensureBoolean throws error on non boolean", () => {
+  assertThrows(() => ensureBoolean(0));
+  assertThrows(() => ensureBoolean("a"));
+  assertThrows(() => ensureBoolean([]));
+  assertThrows(() => ensureBoolean({}));
+  assertThrows(() => ensureBoolean(function () {}));
+  assertThrows(() => ensureBoolean(undefined));
+  assertThrows(() => ensureBoolean(null));
+});
```
Comment (lambdalisue) [inline ensure_test.ts:64]: Please add tests like the below to all other tests

```typescript
assertThrows(() => ensureXXX(true));
assertThrows(() => ensureXXX(false));
```

File: is_test.ts [hunk 3]
```diff
@@ -172,6 +172,19 @@ Deno.test("isLike returns true/false on number", () => {
...
+  assertEquals(isLike(ref, false), true);
+
+  assertEquals(isLike(ref, 0), false);
+  assertEquals(isLike(ref, ""), false);
+  assertEquals(isLike(ref, []), false);
+  assertEquals(isLike(ref, {}), false);
+  assertEquals(isLike(ref, function () {}), false);
+  assertEquals(isLike(ref, null), false);
+  assertEquals(isLike(ref, undefined), false);
+});
```
Comment (lambdalisue) [inline is_test.ts:187]: Please add the below to other tests

```typescript
assertEquals(isLike(ref, true), false);
assertEquals(isLike(ref, false), false);
```

**Review by lambdalisue — CHANGES_REQUESTED** [review]
Additionally, please add entries to `README.md`

**Review by lambdalisue — APPROVED** [review]
LGTM

**Review by lambdalisue — CHANGES_REQUESTED** [review]
Please fix CI error

**Comment by lambdalisue** [comment]
Additionally, please click "Re-request review" so that I can notice changes. @monaqa 

CleanShot 2021-07-21 at 10 22 29

**Comment by monaqa** [comment]
Thank you for your review, and sorry, I forgot to run `deno fmt`.

**Review by lambdalisue — APPROVED** [review]

**Pull request merged**